core/domain/feedback_services.py

```diff
@@ -291,12 +307,51 @@ def _add_feedback_message_reference(user_id, reference):
         enqueue_feedback_message_email_task(user_id)
 
 
-def send_feedback_message_email(exploration_id, thread_id, message_id):
+def update_feedback_email_retries(user_id):
+    model = feedback_models.UnsentFeedbackEmailModel.get(user_id)
+    time_since_buffered = (
+        (datetime.datetime.utcnow() - model.created_on).seconds)
+
+    if (time_since_buffered >
+            feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_COUNTDOWN_SECS):
+        model.retries += 1
+        model.put()
+
+
+def pop_feedback_message_references(user_id, references_length):
+    """Pop feedback message references which have been processed already.
+
+    Args:
+    - user_id: id of the receiving user.
+    - references_length: no. of feedback message references that have been
+    processed already."""
+    model = feedback_models.UnsentFeedbackEmailModel.get(user_id)
+
+    if references_length == len(model.feedback_message_references):
+        model.delete()
+    else:
+        message_references = (
+            model.feedback_message_references[references_length:])
+        model.delete()
+        # model is delted and re-created so that created_on property of model
```

Comment:
Docstring a bit unclear; how about this?

```
We delete and recreate the model in order to re-initialize its 'created_on' property and reset the retries count to 0. If we don't do this, then the retries count will be incorrect.
```

Please make sure I've got your meaning right.